mws/resource_mws_permission_assignment.go

```diff
@@ -2,131 +2,88 @@ package mws
+		if v.Principal.PrincipalId != principalId {
 			continue
 		}
-		return Permissions{v.Permissions}, nil
+		return iam.UpdateWorkspaceAssignments{Permissions: v.Permissions}, nil
 	}
 	return res, apierr.NotFound(fmt.Sprintf("%d not found", principalId))
 }
 
-func (a PermissionAssignmentAPI) List(workspaceId int64) (list PermissionAssignmentList, err error) {
-	if a.client.Config.AccountID == "" {
-		return list, errors.New("must have `account_id` on provider")
-	}
-	path := fmt.Sprintf("/accounts/%s/workspaces/%d/permissionassignments",
-		a.client.Config.AccountID, workspaceId)
-	err = a.client.Get(a.context, path, nil, &list)
-	return
-}
-
 func ResourceMwsPermissionAssignment() common.Resource {
-	type entity struct {
-		WorkspaceId int64    `json:"workspace_id"`
-		PrincipalId int64    `json:"principal_id"`
-		Permissions []string `json:"permissions" tf:"slice_as_set"`
-	}
-	s := common.StructToSchema(entity{},
-		common.NoCustomize)
+	s := common.StructToSchema(iam.UpdateWorkspaceAssignments{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["workspace_id"] = &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			}
+			m["principal_id"] = &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			}
+			return m
+		})
 	pair := common.NewPairID("workspace_id", "principal_id").Schema(
 		func(m map[string]*schema.Schema) map[string]*schema.Schema {
 			return s
 		})
 	return common.Resource{
 		Schema: s,
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var assignment entity
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
+			var assignment iam.UpdateWorkspaceAssignments
 			common.DataToStructPointer(d, s, &assignment)
-			api := NewPermissionAssignmentAPI(ctx, c)
-			err := api.CreateOrUpdate(assignment.WorkspaceId, assignment.PrincipalId,
-				Permissions{assignment.Permissions})
+			assignment.PrincipalId = common.GetInt64(d, "principal_id")
+			assignment.WorkspaceId = common.GetInt64(d, "workspace_id")
+			_, err = acc.WorkspaceAssignment.Update(ctx, assignment)
 			if err != nil {
 				return err
 			}
 			pair.Pack(d)
 			return nil
 		},
 		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			acc, err := c.AccountClient()
+			if err != nil {
+				return err
+			}
 			workspaceId, principalId, err := pair.Unpack(d)
 			if err != nil {
 				return fmt.Errorf("parse id: %w", err)
 			}
-			list, err := NewPermissionAssignmentAPI(ctx, c).List(common.MustInt64(workspaceId))
+			list, err := acc.WorkspaceAssignment.ListByWorkspaceId(ctx, common.MustInt64(workspaceId))
 			if err != nil {
 				return err
 			}
-			permissions, err := list.ForPrincipal(common.MustInt64(principalId))
+			permissions, err := getPermissionsByPrincipal(*list, common.MustInt64(principalId))
 			if err != nil {
 				return err
 			}
 			return common.StructToData(permissions, s, d)
```

Comment:
We need to set workspace ID and principal ID explicitly - otherwise, import won't work correctly